subgraphs/tornado-cash/src/common/getters.ts

```diff
@@ -0,0 +1,414 @@
+      token.decimals = ETH_DECIMALS;
+    } else if (tokenAddress == TORN_ADDRESS.get(network)!) {
+      token.name = TORN_NAME;
+      token.symbol = TORN_SYMBOL;
+      token.decimals = TORN_DECIMALS;
+    } else if (tokenAddress == Address.fromString(BNB_ADDRESS)) {
+      token.name = BNB_NAME;
+      token.symbol = BNB_SYMBOL;
+      token.decimals = BNB_DECIMALS;
+    } else {
+      token.name = fetchTokenName(tokenAddress);
+      token.symbol = fetchTokenSymbol(tokenAddress);
+      token.decimals = fetchTokenDecimals(tokenAddress) as i32;
+    }
+    token.lastPriceBlockNumber = blockNumber;
+  }
+
+  if (!token.lastPriceUSD || token.lastPriceBlockNumber! < blockNumber) {
+    let price = getUsdPricePerToken(tokenAddress, blockNumber);
+    if (price.reverted) {
+      token.lastPriceUSD = BIGDECIMAL_ZERO;
+    } else {
+      token.lastPriceUSD = price.usdPrice.div(price.decimalsBaseTen);
+    }
+    token.lastPriceBlockNumber = blockNumber;
+  }
+  token.save();
+
+  return token;
+}
+
+export function getOrCreateRewardToken(
+  address: Address,
+  blockNumber: BigInt
+): RewardToken {
+  let rewardToken = RewardToken.load(
+    RewardTokenType.DEPOSIT.concat("-").concat(address.toHexString())
+  );
+
+  if (!rewardToken) {
+    let token = getOrCreateToken(address, blockNumber);
+
+    rewardToken = new RewardToken(
+      RewardTokenType.DEPOSIT.concat("-").concat(address.toHexString())
+    );
+
+    rewardToken.token = token.id;
+    rewardToken.type = RewardTokenType.DEPOSIT;
+
+    rewardToken.save();
+  }
+
+  return rewardToken;
+}
+
+export function getOrCreatePool(
+  poolAddress: string,
+  event: ethereum.Event
+): Pool {
+  let pool = Pool.load(poolAddress);
+
+  let token: Token;
+  if (!pool) {
+    pool = new Pool(poolAddress);
+
+    let network = dataSource.network().toUpperCase();
+
+    let contractTCERC20 = TornadoCashERC20.bind(
+      Address.fromString(poolAddress)
+    );
+    let token_call = contractTCERC20.try_token();
+    if (!token_call.reverted) {
+      token = getOrCreateToken(token_call.value, event.block.number);
+
+      let denomination_call = contractTCERC20.try_denomination();
+      if (!denomination_call.reverted) {
+        let denomination = denomination_call.value;
+
+        pool.name = `TornadoCash ${bigIntToBigDecimal(
+          denomination,
+          token.decimals
+        )}${token.symbol}`;
+        pool.symbol = `${bigIntToBigDecimal(denomination, token.decimals)}${
+          token.symbol
+        }`;
+        pool._denomination = denomination;
+      }
+      pool.inputTokens = [token.id];
+    } else {
+      let tokenAddr: string;
+      let denomination_call: ethereum.CallResult<BigInt>;
+
+      if (network == Network.BSC) {
+        tokenAddr = BNB_ADDRESS;
+        let contractTCBNB = TornadoCashBNB.bind(
+          Address.fromString(poolAddress)
+        );
+        denomination_call = contractTCBNB.try_denomination();
+      } else {
```

Comment:
Will all these values change? If they don't, would it make sense to add them as config?

From a performance point of view it won't matter much since they'll only run once (right?). But I'm more concerned on code complexity and having to handle a different case per network.